sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -887,6 +887,14 @@ object SQLConf {
       .booleanConf
       .createWithDefault(false)
 
+  val IGNORE_NULL_FIELDS_STREAMING_SCHEMA_INFERENCE =
```

Comment:
* I was expecting a configuration for the JSON data source instead of a Spark SQL conf. Say, how to implement this feature if the data source is outside Spark?
* Since Spark SQL cannot control the behavior of external data source implementations, the flag won't work unless external data sources recognize it. This is another reason to put it under JSON data source.

Reply:
ok, I dropped the conf in `SQLConf`.